torch/csrc/jit/export.cpp

```diff
@@ -241,180 +197,546 @@ void encodeBlock(onnx::GraphProto * p_g, Block *b,
-      if (node->kind() == aten::expand) {
-        FAIL_EXPORT(
-            "Could not export a broadcasted operation; ONNX likely does not support this form of broadcasting.\n\nBroadcast occurred at:\n" +
-            getNodeStackTraceString(node));
-      }
-      if (node->kind() == prim::PackPadded || node->kind() == prim::PadPacked) {
-        FAIL_EXPORT(
-            "Cannot export individual pack_padded_sequence or pad_packed_sequence; these operations must occur in pairs.\n\nUsage of this operation occurred at:\n" +
-            getNodeStackTraceString(node));
-      }
-      bool is_aten_fallback = operator_export_type == onnx::OperatorExportTypes::ONNX_ATEN_FALLBACK;
-      if (!node->kind().is_onnx() && !is_aten_fallback && node->kind() != prim::Undefined) {
-        FAIL_EXPORT(
-            "Couldn't export operator " + node->kind().toDisplayString() + "\n\nDefined at:\n" +
-            getNodeStackTraceString(node));
-      }
+          "\n\nDefined at:\n" + getNodeStackTraceString(node));
     IR_END()
-#undef FAIL_EXPORT
   }
+  EncodeBlock(attr_proto->mutable_g(), method->graph()->block(), {});
 }
 
+void ModuleEncoder::EncodeTensor(
+    onnx::TensorProto *tensor_proto,
+    const at::Tensor &tensor,
+    const at::optional<std::string> external_ref) {
+  for(auto d : tensor.sizes()) {
+    tensor_proto->add_dims(d);
+  }
+  tensor_proto->set_data_type(ATenTypeToOnnxType(tensor.type().scalarType()));
+  auto t = tensor.contiguous().toBackend(at::kCPU);
+
+  // The int64_data field stores offset, and strides
+  tensor_proto->add_int64_data(tensor.storage_offset());
+  for (auto &d : tensor.strides()) {
+    tensor_proto->add_int64_data(d);
+  }
+
+  auto storage_ptr = tensor.data_ptr();
+  auto dedup_it = storage_dedup_map_.find(storage_ptr);
+  if (dedup_it != storage_dedup_map_.end()) {
+    tensor_proto->set_doc_string(dedup_it->second);
+  } else {
+    std::string storage_name = "$" + std::to_string(storage_counter_++);
+    tensor_proto->set_doc_string(storage_name);
+    JIT_ASSERT(raw_data_export_map_.count(storage_name) == 0);
+    storage_dedup_map_[storage_ptr] = storage_name;
+    raw_data_export_map_[storage_name] = tensor;
+  }
 }
 
+// Pretty printing
 namespace {
+constexpr char indent_char = ' ';
+constexpr size_t indent_multiplier = 2;
 
-RawDataExportMap ToModelProto(
-    const std::shared_ptr<Graph>& graph,
-    const std::vector<at::Tensor> & initializers,
-    int64_t onnx_opset_version,
-    bool defer_weight_export,
-    onnx::OperatorExportTypes operator_export_type,
-    onnx::ModelProto *model_proto) {
-  if (operator_export_type != onnx::OperatorExportTypes::RAW) {
-    validateGraph(graph, operator_export_type);
+std::string idt(size_t indent) {
+  return std::string(indent * indent_multiplier, indent_char);
+}
+
+std::string nlidt(size_t indent) {
+  return std::string("\n") + idt(indent);
+}
+
+void dump(const onnx::TensorProto& tensor, std::ostream& stream) {
```

Comment:
hm, do we really need the dumping logic now that we don't have nanopb? we could use built in protobuf's TextFormat or DebugString